lusidtools/cocoon/utilities.py

```diff
@@ -1361,7 +1363,7 @@ def generate_time_based_unique_id(time_generator: None):
     timestamp = hex(int(current_time * 10000000.0))
```

Comment:
Can increase this to * 1000000000 (microsends) instead of 10000000 (rounded to 100s of microseconds) if they are running too close to each other.